src/lib/Scenes/Search/Input.tsx

```diff
@@ -44,11 +46,33 @@ export const Input = React.forwardRef<TextInput, InputProps & TextInputProps>(
             <StyledInput
               ref={input}
               placeholderTextColor={color("black60")}
-              onFocus={() => setFocused(true)}
-              onBlur={() => setFocused(false)}
               style={{ flex: 1 }}
-              {...rest as any}
+              {...(rest as any)}
+              onChangeText={text => {
+                setValue(text)
+                rest.onChangeText?.(text)
+              }}
+              onFocus={e => {
+                setFocused(true)
+                rest.onFocus?.(e)
+              }}
+              onBlur={e => {
+                setFocused(false)
+                rest.onBlur?.(e)
+              }}
             />
+            {Boolean(value) && showClearButton && (
+              <TouchableOpacity
+                onPress={() => {
+                  input.current.clear()
+                  setValue("")
+                  rest.onChangeText?.("")
+                }}
+                hitSlop={{ bottom: 40, right: 40, left: 0, top: 40 }}
```

Comment:
Does this define the touchable area of the button? Or the extension of the area around the button to capture taps? This is _super_ nit-picky, but we should use 44x44 points as the smallest touch target, so is this appropriate?

```suggestion
                hitSlop={{ bottom: 44, right: 44, left: 0, top: 44 }}
```

Reply:
hitSlop is the extension area growing outwards from the wrapped element, like margin in css. So this will be 80px high + however many px the icon is already